File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/ExceptionEntryProperty.java
package pt.sapo.gis.trace;

import java.io.PrintWriter;
import java.io.StringWriter;

/**
* @author GIS Team 
 * Entry property to wrap an exception 
 */
public class ExceptionEntryProperty extends EntryProperty {
    
    private Exception exception;

    public Exception getException() {
        return exception;
    }
    
    public ExceptionEntryProperty(Exception e) {
        this.exception = e;
        this.put("exceptionMessage", e.getMessage());
        StringWriter sw = new StringWriter();
        e.printStackTrace(new PrintWriter(sw));
        this.put("exceptionStackTrace", sw.toString());
    }    
}

File: SMI-SDK/.NET SDK/Library/gis/trace/ExceptionEntryProperty.cs
namespace pt.sapo.gis.trace
{
    using System;
    using System.IO;

    /**
    * @author GIS Team 
     * Entry property to wrap an exception 
     */
    public class ExceptionEntryProperty : EntryProperty
    {
        public Exception Exception { get; private set; }

        public ExceptionEntryProperty(Exception e)
        {
            Exception = e;
            this["exceptionMessage"] = e.Message;
            this["exceptionStack"] = e.StackTrace;
        }
    }
}